src/list/list.ts

```diff
@@ -0,0 +1,229 @@
+    let node = this.memory.get(this.firstNode);
+    while (node) {
+      const { next } = node;
+      if (next === this.lastNode) {
+        const { content } = this.memory.get(next);
+        this.lastNode = prevNode.next;
+        node.next = null;
+        this.memory.delete(next);
+        this.length -= 1;
+        if (this.length === 1) {
+          this.lastNode = this.firstNode;
+        }
+        return content;
+      }
+      prevNode = this.memory.get(node.next);
+      node = this.memory.get(next);
+    }
+    return null;
+  }
+
+  public find(searchFunction: SearchFunction<T>, globalSearch = false): T | T[] {
+    if (!this.firstNode) {
+      return [];
+    }
+    const matchContent = [];
+    let node: Nodee<T> = this.memory.get(this.firstNode);
+    while (node) {
+      const { content, next } = node;
+      if (searchFunction(content)) {
+        if (globalSearch) {
+          matchContent.push(content);
+        } else {
+          return content;
+        }
+      }
+      node = this.memory.get(next);
+    }
+    return matchContent;
+  }
+
+  public forEach(callback: (content: T) => void): void {
+    if (this.firstNode) {
+      let node: Nodee<T> = this.memory.get(this.firstNode);
+      while (node) {
+        callback(node.content);
+        node = this.memory.get(node.next);
+      }
+    }
+  }
+
+  public map(callback: Function):List<T> {
+    const mapList = new List<T>();
+    if (!this.firstNode) {
+      return mapList;
+    }
+    let node: Nodee<T> = this.memory.get(this.firstNode);
+    let mappedElement: T;
+    while (node) {
+      if (node.next !== null) {
+        mappedElement = callback(node.content);
+        mapList.add(mappedElement);
+      }
+      node = this.memory.get(node.next);
+    }
+    return mapList;
+  }
+
+  public filter(filterFunction: SearchFunction<T>): List <T> {
+    const filterList = new List<T>();
+    if (!this.firstNode) {
+      return new List<T>();
+    }
+    let node: Nodee<T> = this.memory.get(this.firstNode);
+    while (node) {
+      if (filterFunction(node.content)) {
+        filterList.add(node.content);
+      }
+      node = this.memory.get(node.next);
+    }
+    return filterList;
+  }
+
+  public reducer(init: T, callback: (acc: T, currentVal: T) => T):T {
+    let acc: T = init;
+    if (this.firstNode) {
+      let node: Nodee<T> = this.memory.get(this.firstNode);
+      while (node) {
+        acc = callback(acc, node.content);
+        node = this.memory.get(node.next);
+      }
+    }
+    return acc;
+  }
+
+  public some(searchFunction: SearchFunction<T>):boolean {
+    if (!this.firstNode) {
+      return false;
+    }
+    let node: Nodee<T> = this.memory.get(this.firstNode);
+    while (node) {
+      const { content, next } = node;
+      if (searchFunction(content)) {
+        return true;
+      }
+      node = this.memory.get(next);
+    }
+    return false;
+  }
+
+  public every(searchFunction: SearchFunction<T>):boolean {
```

Comment:
your every function if the first element doesn't match continues iterating the whole list. that is inefficient.
If the searchFunction returns false you should immediately return false.
At the en you return true, you don't need to count the timeAppears, just remove that variable.